src/GuppyDataExplorer/ExplorerButtonGroup/index.jsx

```diff
@@ -796,10 +799,12 @@ Currently, in order to export a File PFB, \`enableLimitedFilePFBExport\` must be
       btnTooltipText = 'Currently you cannot export files with different Data Types. Please choose a single Data Type from the Data Type filter on the left.';
     }
 
+
+
     return (
       <Button
         key={buttonConfig.type}
-        onClick={clickFunc}
+        onClick={() => {if (!this.isRegisterUsersForDownloadEnabled()) {clickFunc()}}}
```

Comment:
would be easier to read and understand if refactor into using ternary operators, e.g.: https://github.com/uc-cdis/data-portal/blob/master/src/Discovery/DiscoveryActionBar.tsx#L303-L306

Reply:
Also, the implementation is applying redirect to login for all button types, not download button only

Which means all export buttons will be affected as well, you should confirm with PM this is aligned with the requirement (MIDRC don't have any export buttons for now, but it may in the future)

Reply:
Updated the logic to redirect only for download buttons.